packages/lesswrong/components/hooks/useOnNavigate.tsx

```diff
@@ -19,7 +19,9 @@ const NavigationEventSender = () => {
       // Check if the path has actually changed
       if (location.pathname !== lastLocation?.pathname) {
         // Don't send the callback on the initial pageload, only on post-load navigations
-        if (lastLocation) {
+        // Also suppress callbacks when a modal just opened and pushed a dialog URL
+        const suppressForDialogOpen = typeof window !== 'undefined' && !!(window.history?.state?.dialogOpen);
```

Comment:
You don't need the window check, this is running inside of a `useEffect` (and an `if (isClient)` block, which I think is totally redundant given the `useEffect`).

Reply:
Also what bad thing is this preventing?

Reply:
fixed